guides/common/modules/proc_booting-virtual-machines.adoc

```diff
@@ -48,19 +48,14 @@ If you want to change the default values in the template, clone the template and
 . Set *PXE Loader* to *iPXE Embedded*.
 . Select the *Templates* tab.
 . From the *iPXE template* list, select *Review* to verify that the *Kickstart default iPXE* template is the correct template.
-. Configure the `dhcpd.conf` file as follows:
-+
+
+On {ProjectServer}, run:
 [options="nowrap" subs="+quotes,attributes"]
 ----
-if exists user-class and option user-class = "iPXE" {
-  filename "http://{smartproxy-example-com}:8000/unattended/iPXE?bootstrap=1";
-} # elseif existing statements if non-iPXE environment should be preserved
+{foreman-installer} --foreman-proxy-dhcp-ipxefilename "http://{foreman-example-com}/unattended/iPXE?bootstrap=1"
 ----
-+
-If you use an isolated network, use a {SmartProxyServer} URL with TCP port `8000`, instead of the URL of {ProjectServer}.
-+
+
 [NOTE]
 ====
-If you have changed the port using the `--foreman-proxy-http-port installer` option, use your custom port.
-You must update the `/etc/dhcp/dhcpd.conf` file after every upgrade.
-====
+Ensure that `--foreman-trusted-proxies` is configured correctly on the {ProjectServer}.
```

Comment:
At some point we really need some proper documentation for this. This instruction is not actionable for a user.

Reply:
I think now is a good opportunity to reword this. @ekohl could we instead say somehting like: 
```
. Ensure that <complete me here> is configured as trusted proxy on {Project}.
On {ProjectServer}, run:
<command that marks proxy as trusted in it isn't already>
```
Is there any command that could do this? We could add this as a step and drop the note.

Reply:
We already have it described in https://github.com/theforeman/foreman-documentation/blob/master/guides/common/modules/proc_using-vmware-cloud-init-and-userdata-templates-for-provisioning.adoc (at the end, can't link directly) and https://github.com/theforeman/foreman-documentation/blob/master/guides/common/modules/proc_provisioning-ubuntu-autoinstall-through-smart-proxies.adoc.

One thing to know is that foreman-trusted-proxies describes HTTP reverse proxies. A Smart Proxy (AKA Foreman Proxy) can behave as an HTTP reverse proxy when the templates feature is present. Additionally, on a Content Proxy (AKA Capsule) there is another HTTP reverse proxy (in the form of Apache).

So What in fact should happen is that whenever you install a Smart Proxy with Content (AKA Capsule) or a Smart Proxy with Templates that its IP **must** be in the foreman-trusted-proxies setting. Either the IP itself or the range it belongs to. In addition to that, the localhost IPs must be there too (we should create an issue to fix that at the root). It should also be considered that the ProjectServer can have a Smart Proxy with Templates and in that the ProjectServer's external IP must be added.

Whenever you set `--foreman-trusted-proxies` is passed to the installer, all previous values are wiped so the list must be complete.

I know this is terrible from a user's perspective, but it's even worse that we don't really describe this as part of the installation procedures.

Reply:
Wiping all the previous values is really unfortunate. Is there a config file where the value could be appended as a workaround? Either way, I see this is a change worth its own PR, so @mjivraja you can disregard my comment above.

Reply:
>  Is there a config file where the value could be appended as a workaround?

Sadly not that I know of. We should capture it better so I've opened https://github.com/theforeman/foreman-documentation/issues/1850